iris-mpc-cpu/src/hnsw/graph/neighborhood.rs

```diff
@@ -187,28 +143,76 @@ impl<Vector: Clone, Distance: Clone> SortedNeighborhood<Vector, Distance> {
 
     /// Insert the given unsorted list `vals` of new weighted edges into this
     /// sorted neighborhood using a parallelized quicksort algorithm.
-    async fn quicksort_insert<V>(
+    async fn quicksort_insert(
         &mut self,
         store: &mut V,
-        vals: &[(Vector, Distance)],
-    ) -> Result<()>
-    where
-        V: VectorStore<VectorRef = Vector, DistanceRef = Distance>,
-    {
+        vals: &[(V::VectorRef, V::DistanceRef)],
+    ) -> Result<()> {
         let sorted_prefix_size = self.edges.len();
 
         self.edges.extend_from_slice(vals);
         let mut buffer = self.edges.clone();
 
         apply_quicksort(store, &mut self.edges, &mut buffer, sorted_prefix_size).await
     }
+}
+
+impl<V: VectorStore> AsRef<[(V::VectorRef, V::DistanceRef)]> for SortedNeighborhood<V> {
```

Comment:
Could it work with `Deref` instead of `AsRef`? This way we won’t need all the `.as_ref()`.